conjure/servicewriter.go

```diff
@@ -811,6 +811,37 @@ func serviceStructMethodBodyAST(endpointDefinition spec.EndpointDefinition, retu
 
 	if returnsBinary {
 		// if endpoint returns binary, return body of response directly
+		isOptional, err := isReturnTypeSpecificType(endpointDefinition.Returns, visitors.IsOptional)
+		if err != nil {
+			return nil, err
+		}
+		if isOptional {
```

Comment:
Would add a comment before this block that explains that if endpoint returns optional of binary, then status code is checked and `nil` is returned if it is `StatusNoContent`